openeo/extra/job_management/__init__.py

```diff
@@ -598,26 +627,88 @@ def _launch_job(self, start_job, df, i, backend_name, stats: Optional[dict] = No
+                                bearer_token=job_con.auth.bearer if isinstance(job_con.auth, BearerAuth) else None,
+                                job_id=job.job_id,
+                            )
+                            _log.info(f"Submitting task {task} to thread pool")
+                            self._worker_pool.submit_task(task)
+
+                            stats["job_queued_for_start"] += 1
+                            df.loc[i, "status"] = "queued_for_start"
                         except OpenEoApiError as e:
-                            _log.error(e)
-                            df.loc[i, "status"] = "start_failed"
-                            stats["job start error"] += 1
+                            _log.info(f"Failed submitting task {task} to thread pool with error: {e}")
+                            df.loc[i, "status"] = "queued_for_start_failed"
+                            stats["job queued for start failed"] += 1
             else:
                 # TODO: what is this "skipping" about actually?
                 df.loc[i, "status"] = "skipped"
                 stats["start_job skipped"] += 1
 
+    def _process_threadworker_updates(
+        self,
+        worker_pool: _JobManagerWorkerThreadPool,
+        job_db: JobDatabaseInterface,
+        stats: dict,
+    ) -> None:
+        """
+        Processes asynchronous job updates from worker threads and applies them to the job database and statistics.
+
+        This wrapper function is responsible for:
+        1. Collecting completed results from the worker thread pool
+        2. applying database updates for each job result
+        3. applying statistics updates
+        4. Handles errors with comprehensive logging
+
+        :param worker_pool:
+            Thread pool instance managing the asynchronous job operations.
+            Should provide a `process_futures()` method returning completed job results.
+
+        :param job_db:
+            Job database implementing the :py:class:`JobDatabaseInterface` interface.
+            Used to persist job status updates and metadata.
+            Must support the `_update_row(job_id: str, updates: dict)` method.
+
+        :param stats:
+            Dictionary tracking operational statistics that will be updated in-place.
+            Expected to handle string keys with integer values.
+            Statistics will be updated with counts from completed job results.
+
+        :return:
+            None: All updates are applied in-place to the job_db and stats parameters.
+        """
+        results, _ = worker_pool.process_futures()
+        stats_updates = collections.defaultdict(int)
+
+        for result in results:
+            try:
+                # Handle job database updates
+                if result.db_update:
+                    _log.debug(f"Processing update for job {result.job_id}")
+                    job_db._update_row(job_id=result.job_id, updates=result.db_update)
```

Comment:
`_update_row` is not defined on general `JobDatabaseInterface`, only on `FullDataFrameJobDatabase`, so this is going to break `STACAPIJobDatabase` workflows